### ISSUE TITLE
Comma in template number format doesn't work

### ISSUE THREAD

**Issue by floribon** [vega/vega#298]
Using the template `{{datum.value|number:'.2f'}}` works and transforms `1000.123` in `1000.12`

Using the template `{{datum.value|number:',.2f'}}` doesn't work at all, the thousands are not comma separated, and the floating precision is not taking into account either.

Here is a spec ready to run that shows the problem:

```
{
  "width": 400,
  "height": 400,
  "data": [
    {
      "name": "table",
      "values": [12000.123, 23000.123, 47000.123, 52000.123],
      "transform": [{"type": "pie", "field": "data"}]
    }
  ],
  "marks": [
    {
      "type": "arc",
      "from": {"data": "table"},
      "properties": {
        "enter": {
          "x": {"field": {"group": "width"}, "mult": 0.5},
          "y": {"field": {"group": "height"}, "mult": 0.5},
          "startAngle": {"field": "layout_start"},
          "endAngle": {"field": "layout_end"},
          "innerRadius": {"value": 20},
          "outerRadius": {"value": 100},
          "stroke": {"value": "#fff"},
          "fill": {"value": "#ccc"}
        }
      }
    },
    {
      "type": "text",
      "from": {"data": "table"},
      "properties": {
        "enter": {
          "x": {"field": {"group": "width"}, "mult": 0.5},
          "y": {"field": {"group": "height"}, "mult": 0.5},
          "radius": {"value": 100, "offset": 10},
          "theta": {"field": "layout_mid"},
          "fill": {"value": "#000"},
          "align": {"value": "center"},
          "baseline": {"value": "middle"},
          "text": {"template": "ok: {{datum.data|number:'.2f'}}"}
        }
      }
    },
    {
      "type": "text",
      "from": {"data": "table"},
      "properties": {
        "enter": {
          "x": {"field": {"group": "width"}, "mult": 0.5},
          "y": {"field": {"group": "height"}, "mult": 0.5, "offset": 10},
          "radius": {"value": 100, "offset": 10},
          "theta": {"field": "layout_mid"},
          "fill": {"value": "#000"},
          "align": {"value": "center"},
          "baseline": {"value": "middle"},
          "text": {"template": "not ok: {{datum.data|number:',.2f'}}"}
        }
      }
    }
  ]
}
```
http://vega.github.io/vega-editor/

**Comment by jheer** [comment]
Thanks! This was due to a bug in datalib's template methods. I just released a fixed version of that, we should get it rolled into a new patch release of Vega soon.

**Comment by arvind** [comment]
Thanks for reporting the issue. Should be fixed with 1218662.
